### PR TITLE
Us032a/us034b/merge

### PR DESCRIPTION
**Pull request by oddjobz**
Ok, release 0.2 is a clone of master, so this is to merge Lewis's amended changes into 0.2. Once this is done, we can revert yesterday's PR against master.

### CODE REVIEW

File: swagger_server/controllers/session_manager.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+def session_scope():
+    session = ons_env.db.session
+    try:
+        yield session
+        session.commit()
+    except Exception as e:
+        session.rollback()
+        raise SessionScopeException('session failed to commit: ' + str(e))
+    finally:
+        session.close()
```
Comment (jonmorphic) [inline swagger_server/controllers/session_manager.py]: session.remove() if it's a scoped session

File: swagger_server/models/business.py [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
 from sqlalchemy.orm import relationship
 from sqlalchemy.types import Integer, String
-from ..configuration import ons_env
 from .associations import instrument_business_table
-from .base import Base
+from ons_ras_common import ons_env
 
 
-class BusinessModel(Base):
+class BusinessModel(ons_env.db.base):
```
Comment (jonmorphic) [inline swagger_server/models/business.py]: I still don't see why it's necessary to attach the model base class to ons_env. Not that it has much to do with this PR.

File: swagger_server/models/survey_responses.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+from uuid import uuid4
+from ons_ras_common import ons_env
+
+
+class SurveyResponseModel(ons_env.db.base):
+    """
+        This models the 'survey response' table
+    """
+    __tablename__ = 'survey_responses'
+    __table_args__ = None if not ons_env.get('db_schema') else {'schema': ons_env.get('db_schema')}
```
Comment (jonmorphic) [inline swagger_server/models/survey_responses.py]: I thought we'd done away with the need for this schema hack?

File: swagger_server/test/test_respondent_controller.py [hunk 4]
```diff
@@ -1,39 +1,39 @@
...
+        # When the file is posted to the end point with a case_id
+        response = self.client.open(END_POINT_URL + '{}'.format(case_id),
+                                    method='POST', data=data, content_type='multipart/form-data')
+
+        # Then the file uploads successfully
+        self.assert200(response)
+        self.assertEquals(UPLOAD_SUCCESSFUL, response.data.decode("utf-8"))
+
+    def test_survey_responses_case_id_get(self):
+        # Given a survey response is in the database
```
Comment (jonmorphic) [inline swagger_server/test/test_respondent_controller.py]: Like the given-when-then. 👍

File: Procfile [hunk 5]
```diff
@@ -1 +1 @@
-web: ONS_ENV=production python3 -m swagger_server
+web: ONS_ENV=dev python3 -m swagger_server
```
Comment (oddjobz) [inline Procfile]: Needed if not run through Jenkins.

File: manifest.yml [hunk 6]
```diff
@@ -1,8 +1,8 @@
 ---
 applications:
-- name: ras-collection-instrument-dev
+- name: collection-instrument-demo
   memory: 512M
-  host: ras-collection-instrument-dev
+  host: collection-instrument-demo
   services:
-    - cidsvc-pg-db
+    - ras-postgres
```
Comment (oddjobz) [inline manifest.yml]: Yup, manifest work to be done.

File: scripts/fromtest.sh [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+if ! [ -a .build ] ; then
+	echo "Creating Virtual Environment"
+	virtualenv .build -p python3
+fi
+source .build/bin/activate
+pip install -i https://testpypi.python.org/pypi ons_ras_common --upgrade
```
Comment (oddjobz) [inline scripts/fromtest.sh]: Will be obsoleted shortly.

File: swagger_server/controllers/ciupload_controller.py [hunk 8]
```diff
@@ -1,48 +1,93 @@
...
+    :param file: File name
+    :type file: str
+    :param files: The file to upload
+    :type files: werkzeug.datastructures.FileStorage
 
     :rtype: None
     """
-    return 'do some magic!'
+    uploaded_files = None
+    for index in ['files', 'files[]', 'upfile']:
```
Comment (oddjobz) [inline swagger_server/controllers/ciupload_controller.py]: Only one will be supplied, this isn't new code, not sure why it's popping up here.

File: swagger_server/controllers/collectioninstrument.py [hunk 9]
```diff
@@ -245,6 +262,8 @@ def upload(self, exercise_id, fileobject, ru_ref=None, survey_id=DEFAULT_SURVEY)
         blob = fileobject.read()
         size = len(blob)
         blob = ons_env.cipher.encrypt(blob)
+        if '.' in ru_ref:
```
Comment (oddjobz) [inline swagger_server/controllers/collectioninstrument.py]: Yup, will be removed in next push.

File: scripts/test.sh [hunk 10]
```diff
@@ -4,5 +4,5 @@ if ! [ -a .build-test ] ; then
 	virtualenv .build-test -p python3
 fi
 source .build-test/bin/activate
-pip3 -q install -r test-requirements.txt
-pytest --cov=swagger_server/controllers_local --cov-report term-missing
+pip -q install -r test-requirements.txt
+pytest --cov=swagger_server/controllers --cov-report term-missing
```
Comment (oddjobz) [inline scripts/test.sh]: No particular reason.

File: swagger_server/controllers/helper.py [hunk 11]
```diff
@@ -0,0 +1,8 @@
+def is_valid_file_format(file_name, extensions):
+    extensions_tuple = tuple([e for e in extensions.split(",")])
```
Comment (oddjobz) [inline swagger_server/controllers/helper.py]: Would agree.

File: swagger_server/controllers/session_manager.py (hunk 1 above)
Comment (oddjobz) [inline swagger_server/controllers/session_manager.py]: Need to point this one at Lewis.

File: swagger_server/models/business.py (hunk 2 above)
Comment (oddjobz) [inline swagger_server/models/business.py]: It's cleaner.

**Pull request merged**